### PR TITLE
[CPDLP-2966] Improve NPQ application seed data

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

Ticket: https://dfedigital.atlassian.net/browse/CPDLP-2966

We currently have existing seed data in NPQ for applications. However is it adequate for testing. We can beef it up to make it easier for Shahad/Fraser to test things currently in NPQ separation branch.

### Changes proposed in this pull request

Improve NPQ application seed data.

### CODE REVIEW

File: db/seeds/base/add_applications.rb [hunk 1]
```diff
@@ -1,27 +1,60 @@
...
-  :with_random_user,
-  :with_random_work_setting,
-  ecf_id: SecureRandom.uuid,
-  lead_provider: all_lead_providers.last,
-  course: Course.all.sample,
-  lead_provider_approval_status: "accepted",
-  participant_outcome_state: "passed",
-)
+LeadProvider.find_each do |lead_provider|
+  [1, 2, 3, 4].each do
```
Comment (peteryates) [inline db/seeds/base/add_applications.rb:4]: Can we perhaps tone this down a little bit for non-review environments?

I did this on ECF with [a config file](https://github.com/DFE-Digital/early-careers-framework/blob/main/config/seed_quantities.yml) but maybe here just do something like:

```suggestion
LeadProvider.find_each do |lead_provider|
    quantity = (Rails.env.review?) ? 4 : 1
    
    quantity.times do
```

Comment (peteryates) [reply db/seeds/base/add_applications.rb:4]: Thinking about this further, we probably want 0 for `test`

```suggestion
LeadProvider.find_each do |lead_provider|
  quantity = { "review" => 4, "development" => 1 }.fetch(Rails.env, 0)
  
  quantity.times do
```

File: db/seeds/base/add_applications.rb [hunk 2]
```diff
@@ -1,27 +1,62 @@
...
+    FactoryBot.create_list(
+      :application,
+      4,
+      :with_random_user,
+      :with_random_work_setting,
+      ecf_id: SecureRandom.uuid,
+      lead_provider:,
+      course: Course.all.sample,
+      lead_provider_approval_status: "rejected",
+      participant_outcome_state: "passed",
```
Comment (cwrw) [inline db/seeds/base/add_applications.rb:58]: Curious about this value, is this related to our outcomes from ECF? If it is maybe we set it to false on rejected?

**Review by cwrw — APPROVED** [review]

**Review by cwrw — APPROVED** [review]
